### PR TITLE
feat(addons): add Traefik addon

### LINKED ISSUE #584: [enhancement] Add support for traefik v2
**Is your feature request related to a problem? Please describe.**
We're lacking flexibility regarding ingress controllers.

**Describe the solution you'd like**
Implement an addon for the traefik controller. [Traefik][traefik-main]

[traefik-main]: https://doc.traefik.io/traefik/providers/kubernetes-ingress/

**Additional context**

The `traefik` controller supports both `CRD`-based configuration (using the `IngressRoute` resource) or the standard `Ingress` specification.

- [kubernetes-ingress](https://doc.traefik.io/traefik/providers/kubernetes-ingress/)
- [kubernetes-crd](https://doc.traefik.io/traefik/providers/kubernetes-crd/)

**Todos**

- [x] Implement the traefik addon
- [x] Add support for the CRDs (maybe bootstrap the providers ?)

**In the future**
- Implement a network policy from the control plane if https://github.com/traefik/traefik/issues/7379 gets implemented (currently marked as `priority/P2`)

### PR DESCRIPTION
**Pull request by tbobm**
# Add the Traefik addon

## Description

Include the Traefik Helm Chart to deploy `traefik` as an `IngressController`.

Proper installation can be validated using:
```console
$ k port-forward deploy/traefik 9000:9000
Forwarding from 127.0.0.1:9000 -> 9000
Forwarding from [::1]:9000 -> 9000
Handling connection for 9000
Handling connection for 9000
```

And then heading over to the http://localhost:9000/dashboard/ URL.

Closes #584

### Checklist

- [x] CI tests are passing
- [x] README.md has been updated after any changes to variables and outputs. See https://github.com/particuleio/terraform-kubernetes-addons/#doc-generation

### CODE REVIEW

File: traefik.tf [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+          }
+        }
+      }
+    }
+
+    policy_types = ["Ingress"]
+  }
+}
+
+resource "kubernetes_network_policy" "traefik_allow_control_plane" {
```
[the remark this answers is not in the document]
Comment (ArchiFleKs) [reply traefik.tf:136]: Yes if there is no webhook this can be removed

**Comment by ArchiFleKs** [comment]
I think the only one missing now is the allow ingress

File: traefik.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (tbobm) [reply traefik.tf]: removed and historized!

**Comment by ArchiFleKs** [comment]
@tbobm this seems ok right ? Have you tried applying whith traefik disable also ?

**Comment by tbobm** [comment]
Yes @ArchiFleKs , I tried it with `enabled=false` and tested with `ingress_cidrs` set :)

**Comment by ArchiFleKs** [comment]
@tbobm alright, could you rebase ? I see a lot of unrelated commit in the Pr

**Comment by tbobm** [comment]
@ArchiFleKs we should by good now, I messed up my previous rebase

**Review by ArchiFleKs — APPROVED** [review]

**Pull request merged**